rustc_tools_util/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+#![feature(tool_lints)]
+
+use std::env;
+
+#[allow(clippy::useless_let_if_seq)]
+#[macro_export]
+macro_rules! get_VersionInfo {
+    () => {{
+        let major = env!("CARGO_PKG_VERSION_MAJOR").parse::<u8>().unwrap();
+        let minor = env!("CARGO_PKG_VERSION_MINOR").parse::<u8>().unwrap();
+        let patch = env!("CARGO_PKG_VERSION_PATCH").parse::<u16>().unwrap();
+
+        let host_compiler = $crate::get_channel();
+        let commit_hash = option_env!("GIT_HASH").map(|s| s.to_string());
+        let commit_date = option_env!("COMMIT_DATE").map(|s| s.to_string());
+
+        VersionInfo {
+            major,
+            minor,
+            patch,
+            host_compiler,
+            commit_hash,
+            commit_date,
+        }
+    }};
+}
+
+// some code taken and adapted from RLS and cargo
+pub struct VersionInfo {
+    pub major: u8,
+    pub minor: u8,
+    pub patch: u16,
+    pub host_compiler: Option<String>,
+    pub commit_hash: Option<String>,
+    pub commit_date: Option<String>,
+}
+
+// https://internals.rust-lang.org/t/macro-trait-implementation/5492
+impl VersionInfo {
```

Comment:
isn't this function and the `Default` impl essentially always wrong? And also unused, so it could be removed entirely

Reply:
I added the ````Default```` because of ```` #[warn(clippy::new_without_default_derive)] on by default```` :laughing: 

I'll remove the ````::new()```` and ````::default()````